shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/statement/ShardingSpherePreparedStatement.java

```diff
@@ -367,7 +367,8 @@ private ExecutionContext createExecutionContext() {
     private LogicSQL createLogicSQL() {
         List<Object> parameters = new ArrayList<>(getParameters());
         ShardingSphereSchema schema = metaDataContexts.getDefaultMetaData().getSchema();
-        SQLStatementContext<?> sqlStatementContext = SQLStatementContextFactory.newInstance(schema, parameters, sqlStatement);
+        SQLStatementContext<?> sqlStatementContext = SQLStatementContextFactory.newInstance(parameters, sqlStatement);
+        sqlStatementContext.initSchemaBasedContext(schema);
```

Comment:
Is it necessary to use `initSchemaBasedContext` instead of `SQLStatementContextFactory.newInstance`?

Reply:
> Is it necessary to use `initSchemaBasedContext` instead of `SQLStatementContextFactory.newInstance`?

@tristaZero Because it is currently dependent on the schema when initializing the context, but the schema is not clear when using the context to extract the schema, so the context initialization needs to be split into two steps. The first step is to initialize the context independent of the schema, and the second step is to initialize the context related to the schema.